src/nova_tz/rsa_key.rs

```diff
@@ -0,0 +1,75 @@
+use crate::nova_tz::keyblob;
+use crate::nova_tz::Error;
+use crate::nova_tz::Result;
+use dynfmt::{Format, SimpleCurlyFormat};
+use std::convert::TryFrom;
+use std::fs;
+use std::path::Path;
+use uuid::Uuid;
+
+pub struct RsaKey {
+    key_name: String,
+    key_blob: keyblob::KeyBlob,
+}
+
+impl RsaKey {
+    const NOVA_RSA_ADD_PATH: &'static str = "/sys/nova-rsa/add";
+    const NOVA_RSA_REMOVE_PATH: &'static str = "/sys/nova-rsa/remove";
+    const NOVA_RSA_SIGN_PATH: &'static str = "/sys/nova-rsa/keys/{}/sign";
+    const NOVA_RSA_BLOB_PATH: &'static str = "/sys/nova-rsa/keys/{}/key";
+
+    pub fn new(key_path: &str) -> Result<Self> {
+        let key_name = Uuid::new_v4().to_string();
+
+        if !Path::new(key_path).exists() {
+            return Err(Error::BadKeyPath(key_path.to_string()).into());
+        }
+
+        let key_blob_data =
+            fs::read(key_path).map_err(|_| Error::BadKeyPath(key_path.to_string()))?;
```

Comment:
Why mask io errors here?

Reply:
I'm still not a fan of ignoring the io::Error and converting it into a string here.